### DISCUSSION TITLE
Unity Catalog localhost setup

### DISCUSSION THREAD

**Discussion by MrPowers** [unitycatalog/unitycatalog#264] [Ideas]
I'd like to register all of my localhost data assets in a Unity Catalog instance, so I can easily remember where they are all stored.

Is there a way for me to create a little UC database locally and then point the localhost server to my localhost database?

**Comment by VillePuuska** [comment]
Thanks to PR #211 you can simply specify the location of your database file in [etc/conf/hibernate.properties](https://github.com/unitycatalog/unitycatalog/blob/main/etc/conf/hibernate.properties), or of course use MySQL or Postgres instead of H2 if you'd prefer.

Someone correct me if I'm wrong, but I think if you want to create a completely new H2 db instead of using the default one you can just specify a new path for the db file and that's enough; no steps needed to create it beforehand.

**Reply by jaceklaskowski** [reply]
Think you've just planted a new demo idea into my head 😉  I'm gonna give this switch a serious try! ❤️

**Reply by MrPowers** [reply]
@jaceklaskowski - can you let me know when you figure it out 👼 

My dream setup is a script that will create all the tables I want for localhost experimentation with a single command and have them all neatly organized in a Unity Catalog.

**Reply by VillePuuska** [reply]
Related to having a script create tables and register them to UC, I've been playing around with the idea of having a dataframe API for interacting with UC in this toy project https://github.com/VillePuuska/Local-Lakehouse This basically lets you
- take existing files/tables (Delta, Parquet, CSV, Avro) and register them as tables in UC,
- create tables from Polars dataframes,
- and read/scan tables from UC to Polars dataframes/lazyframes,

without needing to "manually" interact with UC.

[This mini tutorial](https://github.com/VillePuuska/Local-Lakehouse/blob/main/mini_tutorial.py) shows the idea better than I can explain 😄 With this it should be very simple to write a script that builds up a local experimentation catalog.

Of course it's probably possible to use the Spark integration and write a PySpark script that would be just as easy 🤔
